Reina - Added changes to allow badge filtering in the badge management page

ReinaT5678 opened the pull request:
# Description
Made the filtering UI for badge management more user friendly by adding several features to filter by name, type, or rank.


## Related PRS (if any):
This frontend PR is related to the development backend PR.

…

## Main changes explained:
- Added a button for additional features.
- Users can also search by name, type, and can filter by rank.
…

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. Clear site data/cache
4. log as owner/admin user
5. go to dashboard→ Other Links → Badge Management → Badge Development → Add Filters
6. Verify that the filter inputs (Name, Type, and Rank) under the Add Filters buttons works as expected

## Screenshots or videos of changes:


https://github.com/user-attachments/assets/599e4d40-3358-400d-b46a-383e49bb2226

RishithaMamidala reviewed (changes requested):
@ReinaT5678 
All the filters works correctly.
But users cannot sort the badges in ascending or descending order according to name and rank as explained in the changes. It is fixed to be in ascending order and cannot be changed to descending..
<img width="926" alt="PR 2671" src="https://github.com/user-attachments/assets/17f36c38-1d28-4cd0-b397-029f86cf71c4">

ReinaT5678 commented:
> @ReinaT5678 All the filters works correctly. But users cannot sort the badges in ascending or descending order according to name and rank as explained in the changes. It is fixed to be in ascending order and cannot be changed to descending.. <img alt="PR 2671" width="926" src="https://private-user-images.githubusercontent.com/113060798/367423956-17f36c38-1d28-4cd0-b397-029f86cf71c4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYyNTc3MTUsIm5iZiI6MTcyNjI1NzQxNSwicGF0aCI6Ii8xMTMwNjA3OTgvMzY3NDIzOTU2LTE3ZjM2YzM4LTFkMjgtNGNkMC1iMzk3LTAyOWY4NmNmNzFjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkxM1QxOTU2NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOGIzMDFiMjhjNTkzOGJjNzMyOWQ4Y2EzNTNjY2E4YTczOTMxM2MwZjhmM2IyODE4MjUzY2IyNTczMDdhMGFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.dU58mgWTkDZbBDm19od2ylQWlrmPsIFD87lgBX0iWOQ">

Hi, thank you for pointing that out! I removed that feature since it interfered with a header test. I changed the comment to remove that feature.

sam19981 reviewed (approved):
Changes are working as expected.
<img width="1433" alt="Screenshot 2024-09-15 at 12 56 25 AM" src="https://github.com/user-attachments/assets/7a895fba-70af-4eb8-9039-c50200c5770c">

ashayk9 reviewed (approved):
I tested the pr and its working as expected, all filters working fine.
![Screenshot 2024-09-17 at 10 14 03 PM](https://github.com/user-attachments/assets/1a6775a2-afcf-433a-9878-b43ac8feec9e)
![Screenshot 2024-09-17 at 10 14 26 PM](https://github.com/user-attachments/assets/62230e84-0b46-4c78-80a7-b2b598ab2706)
![Screenshot 2024-09-17 at 10 14 51 PM](https://github.com/user-attachments/assets/db9f746a-b7a6-4dd3-a65c-0bb2a834b811)

geetamatkar commented:
Hi @ReinaT5678 ,
Tested this PR and all the 3 filters are working as expected.

<img width="1430" alt="PR2671-3" src="https://github.com/user-attachments/assets/816b7edc-3bc6-43bd-8e28-c4ed12549331">
<img width="1395" alt="PR2671-2" src="https://github.com/user-attachments/assets/87338abe-531b-4693-9774-454e89fc3c47">
<img width="1450" alt="PR2671-1" src="https://github.com/user-attachments/assets/4f64269a-5796-4b09-b07c-d79a99c4d2fd">

chu-ziyu reviewed (approved):
I tested this PR and filter worked as expected in both light mode and dark mode.
![Screenshot 2024-09-18 at 5 55 03 PM](https://github.com/user-attachments/assets/676b9ebe-440f-476c-a159-d6db0747079b)
![Screenshot 2024-09-18 at 5 56 24 PM](https://github.com/user-attachments/assets/3ed4067a-4241-47a2-9555-ec37ff936f16)

Lexie5212 reviewed (dismissed):
I tested this PR , and the filters of name, type and rank worked.

https://github.com/user-attachments/assets/07b06319-5bc3-4fbe-b5b1-7a557a00a162

tharunaa-1 commented:
Expected result after adding the filter, works perfectly.
![image](https://github.com/user-attachments/assets/fbf5e377-0d54-4fd0-a117-282e2b7f6819)

![image](https://github.com/user-attachments/assets/0a53c050-61e6-4f4d-9539-3af3832d2df2)

Muzammil603 reviewed (approved):
changes are working as expected
![image](https://github.com/user-attachments/assets/57220441-786d-4618-8177-5972db622338)

ashmita123 reviewed (dismissed):
I tested the badge management filters, and everything is working smoothly. The name, type, and rank filters function as expected. Nice work on this update!

https://github.com/user-attachments/assets/c768a196-07f6-49f3-b21c-51d1e186c9ed

nevatiarishabh reviewed (dismissed):
I have tested the PR review. Everything is working as expected. All the badge management filters are working alright. 

<img width="1440" alt="Screenshot 2024-09-20 at 4 27 26 PM" src="https://github.com/user-attachments/assets/d7a1f51a-213c-463a-a943-53a7c633ad99">

Abhinav-Ankur786 reviewed (approved):
It is working as per PR

<img width="1479" alt="PR -#2671" src="https://github.com/user-attachments/assets/b9f16566-db25-4023-80ec-33bf99126a03">

nishitag5 reviewed (approved):
The changes are working as expected.

https://github.com/user-attachments/assets/d063b0c7-14f4-4bc0-8237-4680ac688910

saurabhshetty29 commented:
Filtered results are working as expected in both light and dark mode.
<img width="1423" alt="Screenshot 2024-09-21 at 11 41 00 AM" src="https://github.com/user-attachments/assets/4b344841-2069-4815-a793-0fa0da24d27c">

nikhilgiri0226 reviewed (approved):
![image](https://github.com/user-attachments/assets/e62675ed-d6c5-45fe-9458-fa4e1a8fc8b0)

![image](https://github.com/user-attachments/assets/36b968ab-b068-41fa-9ea5-901cd54938c2)

Verified that the filter inputs (Name, Type, and Rank) under the Add Filters buttons works as expected

mrinalini-raghavendran19 reviewed (dismissed):
Looks good to me!

This PR works as expected and the user can sort and filter with the given filters appropriately:


https://github.com/user-attachments/assets/1dc4b067-3a3e-43f8-adc2-83059438fa22

SFA23SCM35V reviewed (approved):
Tested the PR by verifying that the filter inputs (Name, Type, and Rank) under the Add Filters button function correctly. All features are working as expected.
![1](https://github.com/user-attachments/assets/7c5123fb-7b6b-4480-a653-5cb3ed1f258f)
![2](https://github.com/user-attachments/assets/57c0e0c0-dcaf-4ccb-96a9-d8fc507448bc)

khushijain03 reviewed (dismissed):
Hi @Reina128, 
Tested the filters, all 3 are working as expected!

<img width="1465" alt="Screenshot 2024-09-26 at 9 30 49 PM" src="https://github.com/user-attachments/assets/97664c8c-9a83-4b89-820a-b25b9c40ac9b">

<img width="1469" alt="Screenshot 2024-09-26 at 9 31 15 PM" src="https://github.com/user-attachments/assets/1b86a28d-e10b-45c0-92c0-7bc68830d8dc">

<img width="1469" alt="Screenshot 2024-09-26 at 9 31 44 PM" src="https://github.com/user-attachments/assets/bfe30de2-f0b5-4268-b267-c605568d005c">

neeharikakoniki reviewed (approved):
This PR effectively adds functional filter inputs for Name, Type, and Rank in the Badge Development section.
![image](https://github.com/user-attachments/assets/972c64bb-f2f0-4e92-a8ba-b3a1c76aabbe)

ziyanw52 reviewed (dismissed):
The filter for name, type, and rank all works.
![311727469817_ pic_hd](https://github.com/user-attachments/assets/dd3d19c5-702e-4267-92ce-f203ba8a8ddc)

Vincent-space reviewed (approved):
Hi, I've reviewed the PR, everything worked as expected.
<img width="1287" alt="image" src="https://github.com/user-attachments/assets/d8520880-0d96-4bdf-9a3a-aa4552c7a870">

pallavithorat commented:
This pull request successfully implements functional filter inputs for Name, Type, and Rank within the Badge Development section.
<img width="1470" alt="PR #2671" src="https://github.com/user-attachments/assets/ced40177-d9bd-4274-ae91-1c7fd7d1c9d9">

Anoushka21 reviewed (dismissed):
Everything looks good- can filter the badge by name, rank, type.
I tested out with a admin account in dark and light mode both. Things look good!


https://github.com/user-attachments/assets/b5cddb20-0380-4aed-9303-44bd70395281

AshishNags reviewed (dismissed):
Approving this PR, filters working as expected.
![Screenshot 2024-10-18 122723](https://github.com/user-attachments/assets/97b772f6-872f-46da-910d-a7327f6de777)
![Screenshot 2024-10-18 123232](https://github.com/user-attachments/assets/3c9e1e1a-e630-47af-933a-4e5ab8320b88)

bhavya17prakash reviewed (approved):
The filter fields are working as expected. 

https://github.com/user-attachments/assets/00aa9d30-14ca-4160-a0bf-b4810a5ee433

VMD281 commented:
I reviewed these changes and it works perfectly.
<img width="950" alt="PR#2671_1" src="https://github.com/user-attachments/assets/7a413614-47e6-44e5-9f03-a60c21269184">

shashankhv reviewed (dismissed):
Reviewed and tested the changes. Everything is working as expected. Approving the PR.

[Badge-Management.webm](https://github.com/user-attachments/assets/ca14b167-6d77-40b1-8b67-e30ec2ea88ff)

zhiminV reviewed (approved):
I have tested the function and the filter inputs (Name, Type, and Rank) under the Add Filters buttons works as expected.


https://github.com/user-attachments/assets/f761a08f-5efe-42e2-ad61-aeef7d69197f

Rupa2400 reviewed (dismissed):
Reviewed and tested PR #2671 

All the filters are working exactly as expected. Good Job!

<img width="959" alt="PR_2671" src="https://github.com/user-attachments/assets/22c8e91c-da57-48d8-bd95-4c585906e838">

https://github.com/user-attachments/assets/c12109f0-29ae-4782-b3f3-28d0d8f80c92

VMD281 commented:
Hi I reviewed this PR and the Badge development function works fine! 
<img width="1710" alt="PR#2671_1" src="https://github.com/user-attachments/assets/44a88926-8647-4f90-8542-2db0a34afbd6">
<img width="1710" alt="PR#2671_2" src="https://github.com/user-attachments/assets/c89124f3-b65c-43b5-ba03-b088bd363d21">

humera314 commented:
Working as expected

![image](https://github.com/user-attachments/assets/b64cffe3-c14a-41a6-8509-de7809dc2930)

SupriyaSudini reviewed (commented):
I have tested all the features - type, rank, range, and name with these filters in both dark and light modes and confirmed that the results are accurate and everything is functioning correctly. 
Additionally, I did not find any button for "additional features" in either mode. Could you please confirm this?

![image](https://github.com/user-attachments/assets/3050f636-db18-447f-8912-6f93fdb0fbd9)

![image](https://github.com/user-attachments/assets/626d9bc4-d6cb-4a6a-aeaa-1f8ad0a8f8c4)

![image](https://github.com/user-attachments/assets/ef9bfddf-4d19-4e1e-ba9c-0e69528ed3f2)

![image](https://github.com/user-attachments/assets/192cb9a3-0edb-46e2-9e3e-55fdea3f227b)

![image](https://github.com/user-attachments/assets/392e44fa-a6dd-4bb3-8de1-ed212b744709)

![image](https://github.com/user-attachments/assets/81ce2e6f-21e2-4f31-9a73-1882ef0dc5be)

AshritaCherlapally reviewed (dismissed):
This PR is working as expected.  Badge Management - Search and Filter by Type are filtering the data as per user request.

https://github.com/user-attachments/assets/840a41fc-2480-4c48-b34e-bc06f2e6963a

vishavk1992 reviewed (approved):
I have tested this Pr and everything works as expected.

one-community commented:
Thank you all, merging!
